### DISCUSSION TITLE
6.x版本，无法应用在laravle10.x中

### DISCUSSION THREAD

**Discussion by jasonencode** [w7corp/easywechat#2723] [General]
PHP版本 8.2.7
Laravel版本：10.15
```shell
 - w7corp/easywechat[6.11.0, ..., 6.12.3] require nyholm/psr7-server ^1.0 -> satisfiable by nyholm/psr7-server[1.0.0, 1.0.1, 1.0.2].
    - w7corp/easywechat[6.0.0, ..., 6.10.0] require monolog/monolog ^2.2 -> found monolog/monolog[2.2.0, ..., 2.9.1] but the package is fixed to 3.4.0 (lock file version) by a partial update and that version does not match. Make sure you list it as an argument for the update command.
    - nyholm/psr7-server 1.0.0 requires php ^7.1 -> your php version (8.2.7) does not satisfy that requirement.
    - nyholm/psr7-server[1.0.1, ..., 1.0.2] require psr/http-message ^1.0 -> found psr/http-message[1.0, 1.0.1, 1.1] but the package is fixed to 2.0 (lock file version) by a partial update and that version does not match. Make sure you list it as an argument for the update command.
    - Root composer.json requires w7corp/easywechat ^6.0 -> satisfiable by w7corp/easywechat[6.0.0, ..., 6.12.3].
```

**Comment by jasonencode** [comment]
加-W 可以解决目前的问题，但是总觉得不是很友好

**Reply by angelofan** [reply]
```
Your requirements could not be resolved to an installable set of packages.

  Problem 1
    - Root composer.json requires w7corp/easywechat 6.7 -> satisfiable by w7corp/easywechat[6.7.0].
    - w7corp/easywechat 6.7.0 requires monolog/monolog ^2.2 -> found monolog/monolog[2.2.0, ..., 2.9.1] but these were not loaded, likely because it conflicts with another require.


Installation failed, reverting ./composer.json and ./composer.lock to their original content.
```

加了 -W 仍然报错，有什么其他解决方法吗？

**Reply by angelofan** [reply]
使用 `composer require w7corp/easywechat:^6.11.3 -W` 可安装成功。

@overtrue  https://easywechat.com/6.x/index.html 中的 `composer require w7corp/easywechat:^6.7` 是否需要更新？

**Reply by overtrue** [reply]
^6.7 代表 >= 6.7.0 && < 7.0.0 

https://getcomposer.org/doc/articles/versions.md

**Comment by overtrue** [comment]
因为这个包不发新版导致的，https://github.com/Nyholm/psr7-server/issues/55

**Reply by angelofan** [reply]
有没有推荐的解决方法？

**Comment by brianping7** [comment]
我也碰到这个问题了，求大神！

**Reply by angelofan** [reply]
`composer require w7corp/easywechat:^6.12.11 -W`

**Reply by brianping7** [reply]
Problem 1
    - Root composer.json requires w7corp/easywechat ^6.12.11, found w7corp/easywechat[6.x-dev] but it does not match your minimum-stability.
报错啦。。。。

**Reply by angelofan** [reply]
试试 `composer require w7corp/easywechat:^6.7 -W`

**Reply by brianping7** [reply]
Problem 1
    - w7corp/easywechat[6.7.0, ..., 6.8.0] require monolog/monolog ^2.2 -> found monolog/monolog[2.2.0, ..., 2.9.2] but these were not loaded, likely because it conflicts with another require.
    - Root composer.json requires w7corp/easywechat ^6.7 -> satisfiable by w7corp/easywechat[6.7.0, ..., 6.8.0].
报monolog版本问题，现在monolog 3.5.0了，降低不下去

**Reply by angelofan** [reply]
试试 `composer require w7corp/easywechat:^6.12.11 -W --no-dev`

**Reply by brianping7** [reply]
好像没有 no-dev这个参数

**Reply by angelofan** [reply]
可以研究一下为啥报 `but it does not match your minimum-stability` 错误。我没搞清楚为啥会匹配到 `[6.x-dev]` 上，查了 https://packagist.org/packages/w7corp/easywechat 看到确实有 `[6.x-dev]` 这个版本，看看怎么能避免匹配到这个版本上面。也就是说，要么解决那个报错，要么找一下怎么不被匹配到dev版本。

**Reply by overtrue** [reply]
因为他用了一个废弃的 packagist 源。

**Reply by rare0619** [reply]
大佬现在除了官方源，还推荐用谁家的国内镜像吗？

**Reply by overtrue** [reply]
> 大佬现在除了官方源，还推荐用谁家的国内镜像吗？

只推荐官方源
